modules/visual_script/visual_script.cpp

```diff
@@ -291,80 +292,113 @@ void VisualScript::_node_ports_changed(int p_id) {
 #endif
 }
 
-void VisualScript::add_node(const StringName &p_func, int p_id, const Ref<VisualScriptNode> &p_node, const Point2 &p_pos) {
+void VisualScript::add_node(const StringName &p_name, int p_id, const Ref<VisualScriptNode> &p_node, const Point2 &p_pos) {
 	ERR_FAIL_COND(instances.size());
-	ERR_FAIL_COND(!functions.has(p_func));
 	ERR_FAIL_COND(p_node.is_null());
 
-	for (Map<StringName, Function>::Element *E = functions.front(); E; E = E->next()) {
-		ERR_FAIL_COND(E->get().nodes.has(p_id)); //id can exist only one in script, even for different functions
+	StringName var_name;
+	if (Object::cast_to<VisualScriptVariableGet>(*p_node)) {
+		Ref<VisualScriptVariableGet> vget = p_node;
+		var_name = vget->get_variable();
+	} else if (Object::cast_to<VisualScriptVariableSet>(*p_node)) {
+		Ref<VisualScriptVariableSet> vset = p_node;
+		var_name = vset->get_variable();
 	}
```

Comment:
p_node can be invalide here.

Reply:
`ERR_FAIL_COND(p_node.is_null());`
This error macro takes care of this